### PR TITLE
Fix uint8 output conversion

### PR DESCRIPTION
**Pull request by leondgarse**
uint8 `output` should be converted according to `output_zero_point` and `output_scale`.

### CODE REVIEW

**Comment by gbaned** [comment]
@leondgarse  Can you please resolve conflicts? Thanks!

**Comment by jianlijianli** [comment]
Hi leondgarse,

Thanks for the PR on post-training quantization. Just wondering what the benefits is for the extra conversion? The "argmax()" works on both 8bit output as well as float output, so the converting back to float seems unnecessary. Right?

**Comment by leondgarse** [comment]
Hi @jianlijianli ,

Ya, you are right. It currently runs fine without converting back to float. My thought is that as a tutorial notebook, we should contain that converting part. This really confused me as I learning how to do this using another model, that it outputs `embedding` value other than `softmax` value.

Please feel free to reject this if you dont like it. :)

**Comment by jianlijianli** [comment]
Understand and I think it worth some clarification.

Instead of converting that back to float, is it possible to just add a comment for "predictions[i] = output.argmax()"? Something alone the line:
"It's a classification so we can call argmax() directly on the quantized output, instead of converting back to float."

Does this make sense? Thanks.

**Comment by leondgarse** [comment]
I finally found out why I make this checkin...
As in the earlier version, it is actually converted to float, like this:
```py
if output_details['dtype'] == np.uint8:
    output_scale, output_zero_point = output_details["quantization"]
    test_image = test_image.astype(np.float32)
    test_image = test_image / input_scale + input_zero_point
```
Which is not right, it should be
```py
if output_details['dtype'] == np.uint8:
    output_scale, output_zero_point = output_details["quantization"]
    output = output.astype(np.float32)
    output = (output - output_zero_point) * output_scale
```
Which is my original checkin...
Then it is just removed in [Fix copy/paste mistake error specifying incorrect dequantization code.](https://github.com/tensorflow/tensorflow/commit/17eb743235f667272865dab7ca944f2a8e706d99#diff-ff5219d777bd073cc5525bcfcc71c24d8667e004ff476a3c25980a2d0df76537)...

In my understanding, even in classification task, converting back to float is essential to get the actual probability value.
```py
# Given the output value of `test_image_indices == 18`, where it's actually `3`, but predicted as `8`
output = np.array([ 99,  47, 142, 163,  79, 161, 116,  61, 176, 110], dtype=uint8)
# Calculate the probability value using this uint8 value, and the max value is `9.9999738e-01`
print(tf.nn.softmax(tf.cast(output, 'float32')).numpy())
# [3.6251316e-34 0.0000000e+00 1.7139039e-15 2.2603235e-06 0.0000000e+00
#  3.0590152e-07 8.7564878e-27 0.0000000e+00 9.9999738e-01 2.1705163e-29]

# The true probability value should be this, where the max value is `8.1651992e-01`
print(tf.nn.softmax((tf.cast(output, 'float32') - output_zero_point) * output_scale).numpy())
# [4.0755808e-06 1.0709489e-09 3.7232505e-03 1.0395887e-01 1.7104284e-07
#  7.5709976e-02 6.0354825e-05 9.8566248e-09 8.1651992e-01 2.3312372e-05]
```
But in this notebook, it just works using `uint8`... So shall I close this?

**Comment by jianlijianli** [comment]
Now I see where you are coming from and your understanding is absolutely right on the numerics. Since now there is no confusion, if you are OK, please feel free to close this.  Thanks.
